chore: fix CodeQL workflow and config

The following is an extract of lint issues seen on CodeQL workflow and config:
```
/home/runner/work/maidr/maidr/.github/codeql/codeql-config.yml
  19:6  error  Parsing error: Map keys must be unique

/home/runner/work/maidr/maidr/.github/workflows/codeql.yml
Error:    1:7   error  Must use plain style scalar   yaml/plain-scalar
Error:    1:7   error  Strings must use singlequote  yaml/quotes
Error:    7:22  error  Must use plain style scalar   yaml/plain-scalar
Error:    7:22  error  Strings must use singlequote  yaml/quotes
Error:    9:18  error  Must use plain style scalar   yaml/plain-scalar
Error:    9:[18](https://github.com/xability/maidr/actions/runs/15375465383/job/43259682440#step:5:19)  error  Strings must use singlequote  yaml/quotes
Error:   43:21  error  Strings must use singlequote  yaml/quotes

✖ 8 problems (8 errors, 0 warnings)
  7 errors and 0 warnings potentially fixable with the `--fix` option.

Error: Process completed with exit code 1.
```

--- a/.github/codeql/codeql-config.yml
+++ b/.github/codeql/codeql-config.yml
@@ -1,4 +1,4 @@
-name: "CodeQL Custom Configuration"
+name: CodeQL Custom Configuration
 
 queries:
   - uses: security-extended
@@ -15,40 +15,31 @@ paths-ignore:
 
 query-filters:
   - include:
-      tags contain: security
-      tags contain: external/cwe/cwe-798   # Hardcoded credentials
-      tags contain: external/cwe/cwe-312   # Cleartext storage of sensitive info
-      tags contain: external/cwe/cwe-522   # Insufficiently protected credentials
-      tags contain: external/cwe/cwe-200   # Information exposure
-      tags contain: external/cwe/cwe-79    # XSS
-      tags contain: external/cwe/cwe-601   # URL Redirection to Untrusted Site
-      tags contain: external/cwe/cwe-201   # Information Exposure Through Sent Data
-      tags contain: external/cwe/cwe-598   # Information Exposure Through Query Strings in GET Request
-      tags contain: external/cwe/cwe-295   # Improper Certificate Validation
-      tags contain: external/cwe/cwe-918   # Server-Side Request Forgery (if relevant)
-      tags contain: external/cwe/cwe-611   # XML External Entity (XXE) Injection (if relevant)
-      tags contain: external/cwe/cwe-89    # SQL Injection (if relevant)
-      tags contain: external/cwe/cwe-20    # Input Validation
-      tags contain: external/cwe/cwe-287   # Improper Authentication
-      tags contain: external/cwe/cwe-285   # Improper Authorization
-      tags contain: external/cwe/cwe-732   # Incorrect Permission Assignment for Critical Resource
-      tags contain: external/cwe/cwe-434   # Unrestricted Upload of File with Dangerous Type
-      tags contain: external/cwe/cwe-502   # Deserialization of Untrusted Data
-      tags contain: external/cwe/cwe-611   # XXE
-      tags contain: external/cwe/cwe-352   # CSRF
-      tags contain: external/cwe/cwe-601   # Open Redirect
-      tags contain: external/cwe/cwe-94    # Code Injection
-      tags contain: external/cwe/cwe-22    # Path Traversal
-      tags contain: external/cwe/cwe-863   # Incorrect Authorization
-      tags contain: external/cwe/cwe-732   # Incorrect Permission Assignment
-      tags contain: external/cwe/cwe-522   # Insufficiently Protected Credentials
-      tags contain: external/cwe/cwe-311   # Missing Encryption of Sensitive Data
-      tags contain: external/cwe/cwe-319   # Cleartext Transmission of Sensitive Information
-      tags contain: external/cwe/cwe-327   # Use of a Broken or Risky Crypto Algorithm
-      tags contain: external/cwe/cwe-330   # Use of Insufficiently Random Values
-      tags contain: external/cwe/cwe-601   # URL Redirection to Untrusted Site
-      tags contain: external/cwe/cwe-918   # SSRF
-      tags contain: external/cwe/cwe-352   # CSRF
-      tags contain: external/cwe/cwe-79    # XSS
-      tags contain: external/cwe/cwe-20    # Input Validation
-      tags contain: external/cwe/cwe-798   # Hardcoded Secrets
+      tags:
+        - security
+        - external/cwe/cwe-798 # Hardcoded credentials
+        - external/cwe/cwe-312 # Cleartext storage of sensitive info
+        - external/cwe/cwe-522 # Insufficiently protected credentials
+        - external/cwe/cwe-200 # Information exposure
+        - external/cwe/cwe-79 # XSS
+        - external/cwe/cwe-601 # URL Redirection to Untrusted Site
+        - external/cwe/cwe-201 # Information Exposure Through Sent Data
+        - external/cwe/cwe-598 # Information Exposure Through Query Strings in GET Request
+        - external/cwe/cwe-295 # Improper Certificate Validation
+        - external/cwe/cwe-918 # Server-Side Request Forgery
+        - external/cwe/cwe-611 # XML External Entity (XXE) Injection
+        - external/cwe/cwe-89 # SQL Injection
+        - external/cwe/cwe-20 # Input Validation
+        - external/cwe/cwe-287 # Improper Authentication
+        - external/cwe/cwe-285 # Improper Authorization
+        - external/cwe/cwe-732 # Incorrect Permission Assignment for Critical Resource
+        - external/cwe/cwe-434 # Unrestricted Upload of File with Dangerous Type
+        - external/cwe/cwe-502 # Deserialization of Untrusted Data
+        - external/cwe/cwe-352 # CSRF
+        - external/cwe/cwe-94 # Code Injection
+        - external/cwe/cwe-22 # Path Traversal
+        - external/cwe/cwe-863 # Incorrect Authorization
+        - external/cwe/cwe-311 # Missing Encryption of Sensitive Data
+        - external/cwe/cwe-319 # Cleartext Transmission of Sensitive Information
+        - external/cwe/cwe-327 # Use of a Broken or Risky Crypto Algorithm
+        - external/cwe/cwe-330 # Use of Insufficiently Random Values

--- a/.github/codeql/codeql-config.yml
+++ b/.github/codeql/codeql-config.yml
@@ -1,8 +1,9 @@
 name: CodeQL Custom Configuration
 
 queries:
-  - uses: security-extended
-  - uses: security-and-quality
+  - uses: security-extended # Broad security coverage
+  - uses: security-and-quality # Code quality checks
+  - uses: ./custom-queries # (Optional) Add project-specific queries later
 
 paths:
   - src
@@ -12,34 +13,3 @@ paths-ignore:
   - '**/node_modules'
   - '**/dist'
   - '**/build'
-
-query-filters:
-  - include:
-      tags:
-        - security
-        - external/cwe/cwe-798 # Hardcoded credentials
-        - external/cwe/cwe-312 # Cleartext storage of sensitive info
-        - external/cwe/cwe-522 # Insufficiently protected credentials
-        - external/cwe/cwe-200 # Information exposure
-        - external/cwe/cwe-79 # XSS
-        - external/cwe/cwe-601 # URL Redirection to Untrusted Site
-        - external/cwe/cwe-201 # Information Exposure Through Sent Data
-        - external/cwe/cwe-598 # Information Exposure Through Query Strings in GET Request
-        - external/cwe/cwe-295 # Improper Certificate Validation
-        - external/cwe/cwe-918 # Server-Side Request Forgery
-        - external/cwe/cwe-611 # XML External Entity (XXE) Injection
-        - external/cwe/cwe-89 # SQL Injection
-        - external/cwe/cwe-20 # Input Validation
-        - external/cwe/cwe-287 # Improper Authentication
-        - external/cwe/cwe-285 # Improper Authorization
-        - external/cwe/cwe-732 # Incorrect Permission Assignment for Critical Resource
-        - external/cwe/cwe-434 # Unrestricted Upload of File with Dangerous Type
-        - external/cwe/cwe-502 # Deserialization of Untrusted Data
-        - external/cwe/cwe-352 # CSRF
-        - external/cwe/cwe-94 # Code Injection
-        - external/cwe/cwe-22 # Path Traversal
-        - external/cwe/cwe-863 # Incorrect Authorization
-        - external/cwe/cwe-311 # Missing Encryption of Sensitive Data
-        - external/cwe/cwe-319 # Cleartext Transmission of Sensitive Information
-        - external/cwe/cwe-327 # Use of a Broken or Risky Crypto Algorithm
-        - external/cwe/cwe-330 # Use of Insufficiently Random Values

--- a/.github/codeql/codeql-config.yml
+++ b/.github/codeql/codeql-config.yml
@@ -3,7 +3,6 @@ name: CodeQL Custom Configuration
 queries:
   - uses: security-extended # Broad security coverage
   - uses: security-and-quality # Code quality checks
-  - uses: ./custom-queries # (Optional) Add project-specific queries later
 
 paths:
   - src

--- a/.github/workflows/codeql.yml
+++ b/.github/workflows/codeql.yml
@@ -38,17 +38,17 @@ jobs:
           fetch-depth: 2
 
       - name: Initialize CodeQL
-        uses: github/codeql-action/init@v2
+        uses: github/codeql-action/init@v3
         with:
           languages: ${{ github.event.inputs.languages }}
-          config-file: ./.github/codeql-config.yml
+          config-file: ./.github/codeql/codeql-config.yml
           queries: +security-extended, security-and-quality # Include additional security queries and override default queries
 
-      - name: Autobuild (if needed)
-        uses: github/codeql-action/autobuild@v2
+      - name: Autobuild
+        uses: github/codeql-action/autobuild@v3
 
       - name: Perform CodeQL Analysis
-        uses: github/codeql-action/analyze@v2
+        uses: github/codeql-action/analyze@v3
         with:
           category: '/language:${{ github.event.inputs.languages }}'
 

--- a/.github/workflows/codeql.yml
+++ b/.github/workflows/codeql.yml
@@ -1,12 +1,12 @@
-name: "CodeQL Analysis"
+name: CodeQL Analysis
 
 on:
   workflow_dispatch:
     inputs:
       languages:
-        description: "Languages to analyze"
+        description: Languages to analyze
         required: true
-        default: "typescript"
+        default: typescript
         type: choice
         options:
           - typescript
@@ -40,7 +40,7 @@ jobs:
       - name: Perform CodeQL Analysis
         uses: github/codeql-action/analyze@v2
         with:
-          category: "/language:${{ matrix.language }}"
+          category: '/language:${{ matrix.language }}'
 
       - name: Create GitHub Issues for Critical Findings
         if: failure()

--- a/.github/workflows/codeql.yml
+++ b/.github/workflows/codeql.yml
@@ -11,7 +11,16 @@ on:
         options:
           - typescript
           - javascript
-          - typescript,javascript
+      severity-level:
+        description: Minimum severity to create issues for
+        required: false
+        default: high
+        type: choice
+        options:
+          - critical
+          - high
+          - medium
+          - low
 
 jobs:
   analyze:
@@ -21,74 +30,97 @@ jobs:
       actions: read
       contents: read
       security-events: write
-      issues: write
-
-    strategy:
-      fail-fast: false
-      matrix:
-        language: ${{ github.event.inputs.languages }}
 
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
 
       - name: Initialize CodeQL
         uses: github/codeql-action/init@v2
         with:
-          languages: ${{ matrix.language }}
+          languages: ${{ github.event.inputs.languages }}
+          config-file: ./.github/codeql-config.yml
+          queries: +security-extended, security-and-quality # Include additional security queries and override default queries
+
+      - name: Autobuild (if needed)
+        uses: github/codeql-action/autobuild@v2
 
       - name: Perform CodeQL Analysis
         uses: github/codeql-action/analyze@v2
         with:
-          category: '/language:${{ matrix.language }}'
+          category: '/language:${{ github.event.inputs.languages }}'
 
-      - name: Create GitHub Issues for Critical Findings
-        if: failure()
+      - name: Create GitHub Issues for Findings
         uses: actions/github-script@v7
         with:
           script: |
-            const fs = require('fs');
-            const path = require('path');
+            // Get the minimum severity from workflow input
+            const minSeverity = '${{ github.event.inputs.severity-level || 'high' }}'.toLowerCase();
+            const severityOrder = ['critical', 'high', 'medium', 'low'];
+            const minSeverityIndex = severityOrder.indexOf(minSeverity);
 
-            // Read the CodeQL results
-            const resultsPath = path.join(process.env.GITHUB_WORKSPACE, 'codeql-analysis-results.json');
-            if (!fs.existsSync(resultsPath)) {
-              console.log('No CodeQL results found');
-              return;
-            }
+            // Get all open CodeQL alerts
+            const { data: alerts } = await github.rest.codeScanning.listAlertsForRepo({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              state: 'open',
+              tool_name: 'CodeQL'
+            });
 
-            const results = JSON.parse(fs.readFileSync(resultsPath, 'utf8'));
+            // Filter alerts by severity
+            const relevantAlerts = alerts.filter(alert => {
+              const alertSeverity = alert.rule.severity.toLowerCase();
+              return severityOrder.indexOf(alertSeverity) <= minSeverityIndex;
+            });
 
-            // Filter for critical findings
-            const criticalFindings = results.filter(result =>
-              result.severity === 'critical' ||
-              result.severity === 'high'
-            );
+            // Create issues for each qualifying alert
+            for (const alert of relevantAlerts) {
+              try {
+                // Check if issue already exists
+                const existingIssues = await github.rest.search.issuesAndPullRequests({
+                  q: `repo:${context.repo.owner}/${context.repo.repo} type:issue in:title "[CodeQL] ${alert.rule.id}"`
+                });
 
-            // Create issues for each critical finding
-            for (const finding of criticalFindings) {
-              await github.rest.issues.create({
-                owner: context.repo.owner,
-                repo: context.repo.repo,
-                title: `[CodeQL] ${finding.rule.id}: ${finding.message}`,
-                body: `
-                  ## CodeQL Security Finding
+                if (existingIssues.data.total_count === 0) {
+                  await github.rest.issues.create({
+                    owner: context.repo.owner,
+                    repo: context.repo.repo,
+                    title: `[CodeQL] ${alert.rule.id}: ${alert.rule.description.substring(0, 100)}`,
+                    body: |
+                      ### CodeQL Security Finding
 
-                  **Severity:** ${finding.severity}
-                  **Rule:** ${finding.rule.id}
-                  **Message:** ${finding.message}
+                      **Severity:** ${alert.rule.severity}
+                      **Rule:** ${alert.rule.id}
+                      **State:** ${alert.state}
 
-                  **Location:** ${finding.location.file}:${finding.location.startLine}
+                      **Location:**
+                      ${alert.most_recent_instance.location.path}:${alert.most_recent_instance.location.start_line}
 
-                  **Description:**
-                  ${finding.rule.description}
+                      **Description:**
+                      ${alert.rule.description}
 
-                  **Recommendation:**
-                  ${finding.rule.recommendation || 'Please review this finding and fix if necessary.'}
+                      **Recommendation:**
+                      ${alert.rule.tags.includes('security') ? 'Security vulnerability - please address immediately.' : 'Please review this finding.'}
 
-                  ---
-                  *This issue was automatically created by CodeQL analysis.*
-                `,
-                labels: ['security', 'codeql', finding.severity]
-              });
+                      **View in Code Scanning:**
+                      ${alert.html_url}
+
+                      ---
+                      *Automatically created by CodeQL analysis (Workflow Run: ${context.runId})*
+                    labels: [
+                      'security',
+                      'codeql',
+                      alert.rule.severity.toLowerCase(),
+                      'automated'
+                    ]
+                  });
+                  console.log(`Created issue for ${alert.rule.id}`);
+                } else {
+                  console.log(`Issue already exists for ${alert.rule.id}`);
+                }
+              } catch (error) {
+                console.error(`Error creating issue for ${alert.rule.id}: ${error.message}`);
+              }
             }

